packages/teams-js/src/private/hostEntity.ts

```diff
@@ -0,0 +1,301 @@
+import { sendMessageToParentAsync } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { ApiName, ApiVersionNumber, getApiVersionTag } from '../internal/telemetry';
+import { ErrorCode, SdkError } from '../public';
+import { TabInformation, TabInstance } from '../public/interfaces';
+import { runtime } from '../public/runtime';
+
+/**
+ * v2 APIs telemetry file: All of APIs in this capability file should send out API version v2 ONLY
+ */
+const hostEntityTelemetryVersionNumber: ApiVersionNumber = ApiVersionNumber.V_2;
+
+/**
+ * @hidden
+ * @internal
+ * @beta
+ * Limited to Microsoft-internal use
+ *
+ * This capability allows an app to associate apps with a host entity, such as a Teams channel or chat, and configure them as needed.
+ */
+export namespace hostEntity {
+  export enum AppTypes {
+    edu = 'EDU',
+  }
+
+  export interface HostEntityIds {
+    /**
+     * Id of the host entity like channel, chat and meetings
+     */
+    threadId: string;
+
+    /**
+     * Id of message in which channel meeting is created
+     */
+    parentMessageId?: string;
```

Comment:
Are the types of these IDs really unbounded strings? Is "Richa" a valid value? Or can they be a more specific type (e.g., a UUID?)

Reply:
We may be able to enforce some kind of regex, but it's definitely not a simple GUID.

Reply:
I think we should let TMP throw incorrect Id related errors because this Id comes from CMD, we don't have any control over it and we don't want to make incorrect assumptions about the format.